internal/provider/validators.go

```diff
@@ -340,6 +340,36 @@ func validateVaultPathsFilter(v interface{}, path cty.Path) diag.Diagnostics {
 	return diagnostics
 }
 
+func validateVaultClusterProxyEndpoint(v interface{}, path cty.Path) diag.Diagnostics {
+	var diagnostics diag.Diagnostics
+
+	err := vaultmodels.HashicorpCloudVault20201125HTTPProxyOption(strings.ToUpper(v.(string))).Validate(strfmt.Default)
+	if err != nil {
+		enumList := regexp.MustCompile(`\[(.*)\]`).FindStringSubmatch(err.Error())
+		expectedEnumList := strings.ToLower(enumList[1])
+
+		// Remove invalid option from allowed list in error message
+		expectedEnumList = strings.ReplaceAll(
+			expectedEnumList,
+			strings.ToLower(string(vaultmodels.HashicorpCloudVault20201125HTTPProxyOptionHTTPPROXYOPTIONINVALID)),
+			"",
+		)
+
+		// Format as comma-separated list
+		expectedEnumList = strings.Join(strings.Fields(expectedEnumList), ", ")
```

Comment:
These steps are extra compared to our other enum validations (for example on the MVU enums). I added this so users don't receive a confusing error message that the invalid/zero enum is actually allowed.

Talked about this a little more in-depth on Michael's PR: https://github.com/hashicorp/terraform-provider-hcp/pull/575#discussion_r1289250842

Reply:
thanks for finding this